### PR TITLE
Log audio selections

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by zzmp** [comment]
### QA Plan

- Open in a fresh install.
- Change your audio device. Ensure that the proper entries are logged in the database.
- Change your mode. Ensure that no entries are logged.
- Change you audio device in new mode. Ensure that the proper entries are logged.
- Change your mode back. Ensure that no entries are logged.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/873/HighFidelity-Beta-PR10747-3d807743176684e54cf30fed457904a64cc5fe79.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/873/HighFidelity-Beta-PR10747-3d807743176684e54cf30fed457904a64cc5fe79.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/876/HighFidelity-Beta-PR10747-4bff925506b2f164b49828e470a86f3c440b5335.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/876/HighFidelity-Beta-PR10747-4bff925506b2f164b49828e470a86f3c440b5335.exe

**Comment by zzmp** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/882/HighFidelity-Beta-PR10747-4bff925506b2f164b49828e470a86f3c440b5335.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/882/HighFidelity-Beta-PR10747-4bff925506b2f164b49828e470a86f3c440b5335.dmg

**Comment by CainFoool** [comment]
Hey @zzmp: Is this ready for QA?

**Comment by CainFoool** [comment]
Testing shortly.

**Comment by CainFoool** [comment]
# Test Failed 

Devices | For these tests, keep the audio menu open; repeat these tests for input and output devices | Input | Output
------------ | ------------ | ------------ | ------------ 
Physically plug/unplug devices | The device lists should update within 2 seconds showing currently connected devices | ✅ | ✅ 
On a fresh install, open Interface in desktop mode | The device should be your Windows default device | ✅ | ✅ 
On a fresh install, open Interface in VR mode | The device should be your VR hardware's device | ✅ | ✅ 
On a fresh install, switch from desktop to VR | The device should switch to your VR hardware's device | It keeps choosing the incorrect microphone. | ✅ 
On a fresh install, switch from VR to desktop | The device should switch to you Windows default device | ✅ | ✅ 
On desktop, select a different device, then... | -
...restart Interface | The device should be your selected device | ✅ | ✅ 
...switch to VR | The device should be your VR hardware's device | It keeps choosing the incorrect microphone. | ✅ 
...switch back to desktop | The device should be your selected device | ✅ | ✅ 
...close Interface, unplug the device, start Interface | The device should be your Windows default device (this must be done last, as it will unselect the device) | ✅ | ✅ 
Repeat by selecting a device in VR | ⬆️, except the desktop device should remain your Windows default | ✅ | ✅ 
Select devices in desktop and VR | ⬆️, except both devices should remain as your selection | ✅  | ✅ 

Let me know if there's anything else I need to do for this.

**Comment by zzmp** [comment]
@CainFoool can you give more details on your issues? Are you using a vive? Does it choose the wrong default device? What is it choosing? What should it be choosing?

**Comment by CainFoool** [comment]
@zzmp:

I have 4 Input Devices and 4 Output Devices currently loaded on my System.

## Inputs

- AT 2020
- Corsair VOID Wireless Microphone
- Logitech C920 Webcam Microphone
- HTC Vive USB Mic

## Outputs
- Speakers
- Corsair VOID Wireless Headset
- HTC Vive USB
- HTC Vive HDMI

I have the Corsair VOID Wireless Headset in both input/output on the SteamVR Settings when it loads up, and the AT2020 Microphone & Speakers as my default Desktop Devices for when not using VR.

When switching between OpenVR and Desktop, the Microphone is choosing the C920 Microphone when switched to VR, and is choosing my normal AT2020 when switching to Desktop.

**Comment by zzmp** [comment]
Does it make the same incorrect choices on master?
If so, this should be a QA Pass, and a separate issue opened up for Vive device selection.

**Comment by zzmp** [comment]
@CainFoool ^

**Comment by CainFoool** [comment]
@zzmp Let me check on Master, give me a few mins.

**Comment by CainFoool** [comment]
@zzmp: Issue appears to be persisting throughout Master, too. QA Pass for 10747.

File: interface/src/scripting/AudioDevices.cpp [hunk 1]
```diff
@@ -43,33 +55,38 @@ QVariant AudioDeviceList::data(const QModelIndex& index, int role) const {
...
-                emit deviceSelected(device.info);
-                emit deviceChanged(device.info);
-            }
-        }
+	return setDevice(index.row(), true);
+}
+
+bool AudioDeviceList::setDevice(int row, bool fromUser) {
+    bool success = false;
+	auto& device = _devices[row];
```
Comment (druiz17) [inline interface/src/scripting/AudioDevices.cpp]: indentation incorrect,

**Comment by zzmp** [comment]
@druiz17 my machine wasn't set up for spaces (over tabs) - a82cc0e should fix that by replacing tabs with spaces. Good catch.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/1052/HighFidelity-Beta-PR10747-a82cc0ec09612035abd906c8de263cdbec94288c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/1052/HighFidelity-Beta-PR10747-a82cc0ec09612035abd906c8de263cdbec94288c.exe

**Pull request merged**